perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2464,19 +2468,42 @@ public Member setSponsoredMember(PerunSession session, Vo vo, User userToBeSpons
 		int processedCounter = 0;
 		while(dataIterator.hasNext()) {
 			Map<String, String> singleRow = dataIterator.next();
-			Map<String, String> singleResult = createSingleSponsoredMemberFromCSV(sess, vo, namespace, singleRow,
-					sponsor, validityTo, sendActivationLink, url, asyncValidation);
-			totalResult.put(data.get(processedCounter++), singleResult);
+			// async validation must be performed at the end, not directly during member creation
+			Validation localValidation = (Objects.equals(Validation.ASYNC, validation)) ? Validation.NONE : validation;
+			Map<String, Object> originalResult = createSingleSponsoredMemberFromCSV(sess, vo, namespace, singleRow,
+					sponsor, validityTo, sendActivationLink, url, localValidation);
+
+			// convert result to expected "type" for outer API
+			Map<String, String> newResult = new HashMap<>();
+			if ("OK".equals(originalResult.get("status"))) {
```

Comment:
All mentioned text representations of the input should be used as constants (to prevent a mistake in copy+paste).

Reply:
It should be fixed.